### PR TITLE
Adds Plugin.Close

### LINKED ISSUE #4: Add Close()
It should auto-generate Close() as below.
https://github.com/tetratelabs/wazero/blob/b1b9b6d40486ca2302de80ae7d3cde05085a76e1/examples/allocation/tinygo/greet.go#L30

### PR DESCRIPTION
**Pull request by codefromthecrypt**
*Issue #, if available:*

Fixes #4

*Description of changes:*

This generates close and changes examples to suggest doing so, as it avoids leaking memory mapped files until GC.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
@mathetake PTAL and if ok I'll merge and cut a release.

File: gen/host.go [hunk 1]
```diff
@@ -147,6 +147,19 @@ func genHost(g *protogen.GeneratedFile, f *fileInfo, service *serviceInfo) {
...
+	g.P(fmt.Sprintf("func (p *%s) Close(ctx %s) (err error) {",
+		pluginName,
+		g.QualifiedGoIdent(contextPackage.Ident("Context")),
+	))
+	g.P(`if r := p.runtime; r != nil {
+				err = r.Close(ctx)
+			}
+			return
+		}`)
+	g.P()
```
Comment (mathetake) [inline gen/host.go:162]: how about adding new line at the end of ^ instead of this?

File: gen/host.go [hunk 2]
```diff
@@ -147,6 +147,19 @@ func genHost(g *protogen.GeneratedFile, f *fileInfo, service *serviceInfo) {
 				config:  config,
 			}, nil
 		}`)
+	g.P()
```
Comment (mathetake) [inline gen/host.go:150]: same

File: gen/host.go [hunk 3]
```diff
@@ -147,6 +147,19 @@ func genHost(g *protogen.GeneratedFile, f *fileInfo, service *serviceInfo) {
...
+
+	// Close plugin
+	g.P(fmt.Sprintf("func (p *%s) Close(ctx %s) (err error) {",
+		pluginName,
+		g.QualifiedGoIdent(contextPackage.Ident("Context")),
+	))
+	g.P(`if r := p.runtime; r != nil {
```
Comment (mathetake) [inline gen/host.go:157]: let's merge these two g.P into single one for readability to see what's going on under Close function

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
thanks for the tips

**Pull request merged**